client/src/retry.rs

```diff
@@ -0,0 +1,122 @@
+//! Turnkey Client to interact with the Turnkey API
+//! See <https://docs.turnkey.com>
+use std::time::Duration;
+
+#[derive(Debug, Clone)]
+pub struct RetryConfig {
+    /// Initial delay before the first retry
+    pub initial_delay: Duration,
+    /// Multiplier for `initial_delay`, yielding the next delay
+    pub multiplier: f64,
+    /// Maximum delay between retries to cap the time between retries.
+    pub max_delay: Duration,
+    /// Maximum number of retries.
+    pub max_retries: usize,
+}
+
+impl Default for RetryConfig {
+    fn default() -> Self {
+        RetryConfig {
+            initial_delay: Duration::from_millis(500),
+            multiplier: 2.0,
+            max_delay: Duration::from_secs(10),
+            max_retries: 5,
+        }
+    }
+}
+
+impl RetryConfig {
+    /// Returns a `RetryConfig` which doesn't allow any retries
+    /// Use this if you do not want the TurnkeyClient to retry on your behalf.
+    /// If you need a retry configuration, look at `::default()`
+    pub fn none() -> Self {
+        RetryConfig {
+            initial_delay: Duration::from_millis(0),
+            multiplier: 0.0,
+            max_delay: Duration::from_secs(0),
+            max_retries: 0,
+        }
+    }
+
+    /// Computes the delay to wait after N attempts have been made.
+    /// `attempt_count` starts at 1 because it never makes sense to wait before the first try.
+    pub fn compute_delay(&self, attempt_count: usize) -> Duration {
+        if attempt_count == 0 {
+            panic!("attempt_count cannot be 0")
```

Comment:
Do we really want to panic!() here? Maybe this can return a Result/Option and be caught by the caller

Reply:
I think we want to `panic` here because this is an incorrect use of the API. I debated adding a Result but I think it's better if the API stays super simple.